### ISSUE TITLE
NXML Reader errors still present

### ISSUE THREAD

**Issue by hickst** [clulab/reach#314]
Still getting same NXML Reader errors on papers PMC2228099 and PMC3018488 despite using latest ai.lum 0.0.5. Please insure that fixes work on these papers, copies of which are available in:
/net/kate/storage/data/nlp/corpora/pmc_openaccess/darpa_eval_mar2016/1krun-error-files_160804

```
==========

 ¡¡¡ NxmlReader error !!!

paper: PMC2228099

error:
java.lang.IllegalArgumentException: requirement failed: children list is empty

stack trace:
scala.Predef$.require(Predef.scala:224)
ai.lum.nxmlreader.standoff.NonTerminal.<init>(Tree.scala:51)
ai.lum.nxmlreader.NxmlDocument.ai$lum$nxmlreader$NxmlDocument$$mkTree$1(NxmlDocument.scala:89)
ai.lum.nxmlreader.NxmlDocument$$anonfun$9.apply(NxmlDocument.scala:98)
ai.lum.nxmlreader.NxmlDocument$$anonfun$9.apply(NxmlDocument.scala:98)
scala.Option.map(Option.scala:146)
ai.lum.nxmlreader.NxmlDocument.mkStandoff(NxmlDocument.scala:98)
ai.lum.nxmlreader.NxmlDocument.<init>(NxmlDocument.scala:103)
ai.lum.nxmlreader.NxmlReader.read(NxmlReader.scala:15)
org.clulab.reach.PaperReader$.getEntryFromPaper(PaperReader.scala:105)
org.clulab.reach.ReachCLI.processPaper(ReachCLI.scala:102)
org.clulab.reach.ReachCLI$$anonfun$2.apply(ReachCLI.scala:52)
org.clulab.reach.ReachCLI$$anonfun$2.apply(ReachCLI.scala:48)
scala.collection.parallel.mutable.ParArray$Map.leaf(ParArray.scala:657)
scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply$mcV$sp(Tasks.scala:49)
scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
scala.collection.parallel.Task$class.tryLeaf(Tasks.scala:51)
scala.collection.parallel.mutable.ParArray$Map.tryLeaf(ParArray.scala:648)
scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask$class.internal(Tasks.scala:159)
scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.internal(Tasks.scala:443)
scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask$class.compute(Tasks.scala:149)
scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:443)
scala.concurrent.forkjoin.RecursiveAction.exec(RecursiveAction.java:160)
scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

**Comment by marcovzla** [comment]
Those papers are read correctly locally. Do you have a custom config file? what is the command you are using?

**Comment by hickst** [comment]
Running ReachCLI in the configuration originally listed in issue #309:

Configuration:
    Bioresources (master) 1.1.15-SNAPSHOT
    Processors (master) 5.9.4-SNAPSHOT w/ build using BR 1.1.15-SNAPSHOT
    Reach (iss285) 1.3.2-SNAPSHOT w/ build using BR 1.1.15-SNAPSHOT and Proc 5.9.4-SNAPSHOT

My application.cnf file is out-of-the-box (unmodified).

I just recompiled Reach, ran those two files alone again and got the same errors.

**Comment by marcovzla** [comment]
I ran it again locally using those snapshots and the default application.conf.
The papers were read successfully:

```
Fri Aug 05 11:34:45 MST 2016: ReachCLI (w/ assembly) begins ...
Fri Aug 05 11:36:04 MST 2016: Starting PMC2096659
Fri Aug 05 11:36:04 MST 2016: Starting PMC2963923
Fri Aug 05 11:36:04 MST 2016: Starting PMC2228099
Fri Aug 05 11:36:04 MST 2016: Starting PMC3018488
Fri Aug 05 11:36:04 MST 2016: Starting PMC3594291
Fri Aug 05 11:36:04 MST 2016: Starting PMC4122675
Fri Aug 05 11:37:02 MST 2016: Finished PMC2228099 successfully (58 seconds)
Fri Aug 05 11:38:50 MST 2016: Finished PMC2096659 successfully (165 seconds)
Fri Aug 05 11:39:45 MST 2016: Finished PMC2963923 successfully (221 seconds)
Fri Aug 05 11:39:54 MST 2016: Finished PMC4122675 successfully (229 seconds)
Fri Aug 05 11:39:59 MST 2016: Finished PMC3018488 successfully (235 seconds)
Fri Aug 05 11:41:00 MST 2016: Finished PMC3594291 successfully (295 seconds)
```

**Comment by hickst** [comment]
?? OK, what am I doing wrong? I will start completely over and retest. Thanks for your help in narrowing this down.

**Comment by hickst** [comment]
Well, I would say that something on my machine is different, but I just ran on Jenny and got the exact same errors as reported above for those 2 files.

**Comment by hickst** [comment]
I then pulled entirely new versions of Bioresources, Processors, and Reach projects from GitHub, recompiled all three (in master branch of each) and got the exact same errors again.

**Comment by hickst** [comment]
Two more notes:
1. I am also cleaning my Ivy local cache before retesting.
2. And, as documented before, I am making Processors use Bioresources 1.1.15-SNAPSHOT:
```
"org.clulab" % "bioresources" % "1.1.15-SNAPSHOT"
```

and making Reach use both local versions:
```
"org.clulab" % "bioresources" % "1.1.15-SNAPSHOT",
"org.clulab" %% "processors" % "5.9.4-SNAPSHOT",
"org.clulab" %% "processors" % "5.9.4-SNAPSHOT" classifier "models",
```

**Comment by hickst** [comment]
I just pulled entirely new versions of Bioresources, Processors, and Reach projects from GitHub, recompiled all three (in master branch of each) *on Jenny* and got the exact same errors again. 
I have now replicated this on 2 different machines with different hardward, with different OSes, and different versions of Java 8. I have to conclude that we are not using the same configuration OR that your setup is caching something that is making the NXML file processing succeed.

**Comment by marcovzla** [comment]
I just tried again. I was using the papers downloaded with Hans' script.
When I use those, everything works, when I use the ones you're using, the system breaks.

I inspected them quickly and I can't tell why one works and the other doesn't. I won't be able to look again until sunday probably.

**Comment by hickst** [comment]
Huh....well it's a clue, at least. Thanks.

**Comment by hickst** [comment]
Given your clue, I reformatted the NXML files (using `tidy -i -xml`) and the processing succeeded! For what it's worth, the report of wc before and after:

#### Before tidy:
```
       3     333    5982 problems/PMC2228099.nxml
       2   10133  176754 problems/PMC3018488.nxml
```

#### After tidy:
```
     207     497    7814 Xproblems/PMC2228099.nxml
    6530   15297  249750 Xproblems/PMC3018488.nxml
```
I don't think explains anything, as **many** of the (successful) input files have only a few, very long lines just like these problematic ones.

**Comment by myedibleenso** [comment]
So `tidy` added newlines?  This is weird... @hickst, could you share a diff?

**Comment by hickst** [comment]
tidy -xml -i will add newlines because the '-i' adds indentation. I only did this as an experiment after I noticed that many input files, including the problematic ones, have only 2 or 3 extremely long lines.

**Comment by marcovzla** [comment]
Is this error still present? I believe this should be fixed in nxmlreader 0.0.6

**Comment by hickst** [comment]
Closed by pull request #323.
